pkg/types/baremetal/platform.go

```diff
@@ -108,4 +108,7 @@ type Platform struct {
 	//
 	// +optional
 	ClusterOSImage string `json:"clusterOSImage,omitempty" validate:"omitempty,osimageuri,urlexist"`
+
+
+	IronicExtraConf map[string]interface{} `json:"ironicExtraConf,omitempty"`
```

Comment:
secondly a map of string to _anything_ is not a great API.

Let's make sure there is an enhancement for this new field before we move ahead on this.

/hold

Reply:
> secondly a map of string to _anything_ is not a great API.
> 
> Let's make sure there is an enhancement for this new field before we move ahead on this.
> 
> /hold

Agreed, mapping to _anything_ is not ideal, the problem is that ironic has ~700 config options and ironic-inspector has another ~400, in order to allow people to deal with special cases we either need to 
1. have some sort of pass through mechanism like this or 
2. add individual options for each one as it comes up with the delay thats involved to merge/release

as is we're seeing patches merged into the ironic image with options specific to CI, specific hardware and for debugging
https://github.com/metal3-io/ironic-image/pull/172
https://github.com/metal3-io/ironic-image/pull/160
https://github.com/metal3-io/ironic-image/pull/167
https://github.com/metal3-io/ironic-image/pull/150

some of these could be top-level platform configuration options but for corner cases I don't think we'd want to start creating options for every single one.

I can create an enhancement for this, I didn't intend on removing the hold on this PR until we came to a consensuses anyways.